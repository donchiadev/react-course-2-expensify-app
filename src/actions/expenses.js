import uuid from 'uuid'

// ADD_EXPENSE action generator
const addExpense = (
    { 
        description = '', 
        note = '', 
        amount = 0,
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
})

// REMOVE_EXPENSE action generator
const removeExpense = ({ id } = { }) => ({
    type: 'REMOVE_EXPENSE',
    id
})

// EDIT_EXPENSE action generator
const editExpense = (id, updates = { }) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
})

export { addExpense, removeExpense, editExpense } 