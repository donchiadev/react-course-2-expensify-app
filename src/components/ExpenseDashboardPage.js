import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseSummary from './ExpensesSummary'

export const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseSummary />
        <ExpenseList />
    </div>
)

export default ExpenseDashboardPage