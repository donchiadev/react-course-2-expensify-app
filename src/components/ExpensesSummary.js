import React from 'react'
import { connect } from "react-redux";
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpenseSummary = (props) => (
    <div>
        <h3>
            {
                props.expensesCount === 1 ? (
                    "There is 1 visible expense."
                ) : (
                    `There are ${props.expensesCount} visible expenses.`
                )
            }
        </h3>
        <h3>Expenses Total is: {numeral(props.expensesTotal / 100).format('$0,0.00')}</h3>
    </div>
)

const mapStateToProps = (state) => {
    const {expenses, filters} = state
    const visibileExpenses = selectExpenses(expenses, filters)
    return {
        expensesCount: visibileExpenses.length,
        expensesTotal: selectExpensesTotal(visibileExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)