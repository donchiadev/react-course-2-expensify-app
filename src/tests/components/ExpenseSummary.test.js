import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpensesSummary'
import selectExpensesTotal from '../../selectors/expenses-total'
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary', () => {
    const wrapper = shallow(<ExpenseSummary />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary with 0 expenses', () => {
    const wrapper = shallow(<ExpenseSummary
        expensesCount={0}
        expensesTotal={0} 
    />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary
        expensesCount={[expenses[0]].length}
        expensesTotal={expenses[0].amount} 
    />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary with expenses', () => {
    const wrapper = shallow(<ExpenseSummary
        expensesCount={expenses.length}
        expensesTotal={selectExpensesTotal(expenses)} 
    />)
    expect(wrapper).toMatchSnapshot()
})