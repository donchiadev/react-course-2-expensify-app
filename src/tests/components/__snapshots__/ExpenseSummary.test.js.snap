// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary 1`] = `
<div>
  <h3>
    There are undefined visible expenses.
  </h3>
  <h3>
    Expenses Total is: 
    $0.00
  </h3>
</div>
`;

exports[`should render ExpenseSummary with 0 expenses 1`] = `
<div>
  <h3>
    There are 0 visible expenses.
  </h3>
  <h3>
    Expenses Total is: 
    $0.00
  </h3>
</div>
`;

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div>
  <h3>
    There is 1 visible expense.
  </h3>
  <h3>
    Expenses Total is: 
    $1,095.00
  </h3>
</div>
`;

exports[`should render ExpenseSummary with expenses 1`] = `
<div>
  <h3>
    There are 3 visible expenses.
  </h3>
  <h3>
    Expenses Total is: 
    $1,945.00
  </h3>
</div>
`;
